### PR TITLE
Build, dev, testing and publishing improvements

### PR DESCRIPTION
**Pull request by smebberson**
It turns out, when we went to `v1.0.0-17.0.0` our testing and publishing processes broke. Testing wasn't properly reporting to codecov and publishing was entirely broken.

This PR resolves those issues.

**_Please note:_** For the moment, the publish step in `codefresh.yml` is running every single time in `--dry-run` mode so you can see what information will be published to NPM.

**Setup**

- [x] Pull down this branch.

**Testing**

- [x] Execute `yarn dev` and make sure you can use the development version of Linz.
- [x] Type `CTRL + c` and make sure the development version stops.
- [x] Execute `yarn dev` and make sure the development version is working again.
- [x] In another terminal window type `yarn test` and make sure the test completes.
- [x] Review all Codefresh builds for this PR, ensure codecov information is being sent to Codecov.
- [x] Review all Codefresh builds for this PR, follow Codecov links ensuring information is on Codecov.
- [x] Review all Codefresh builds for this PR, ensure `npm publish --dry-run` steps contains all required files (and no unnecessary files) for a real `npm publish` in the future.

**Deployment**

- Deprecate `v1.0.0-17.0.0`.
- Republish `v1.0.0-17.1.0`.
- Publish this change as `v1.0.0-17.1.1`.

### CODE REVIEW

**Comment by smebberson** [comment]
@allanchau, this is good to go :)

File: app/lib/redis.js [hunk 1]
```diff
@@ -2,13 +2,11 @@
 
 const Redis = require('ioredis');
 
-Redis.Promise.onPossiblyUnhandledRejection((err) => console.log(`Unhandled Redis error: ${err}`));
```
Comment (allanchau) [inline app/lib/redis.js]: Why did you remove this? I think it catches unhandled promises and prevents the app from crashing?

File: codefresh.yml [hunk 2]
```diff
@@ -43,8 +43,8 @@ steps:
       - /bin/sh
     commands:
       - echo "//registry.npmjs.org/:_authToken=$NPM_AUTH_TOKEN" > ./.npmrc
-      - npm publish
-    when:
-      branch:
-        only:
-          - /^v(?:(?:[0-9]){1,3}\.){2}(?:[0-9]{1,3})-(?:(?:[0-9]){1,3}\.){2}(?:[0-9]{1,3})$/i
+      - npm publish --dry-run
+    # when:
```
Comment (allanchau) [inline codefresh.yml]: Should we just remove these lines?

Comment (allanchau) [reply codefresh.yml]: Oops, ignore this, I just read the comments.
